pandas/io/sql.py

```diff
@@ -617,9 +618,11 @@ def _engine_builder(con):
 
         except ImportError:
             _SQLALCHEMY_INSTALLED = False
+            raise SQLAlchemyRequired("sqlalchemy is required for database URI")
```

Comment:
I personally would just use `ImportError` here. I know we have the custom SQLAlchemyRequired error defined, but it is actually not yet used anywhere.